packages/bpk-component-popover/src/BpkPopoverPortal.js

```diff
@@ -42,21 +44,64 @@ const onOpen = (popoverElement, targetElement) => {
   }
 };
 
-class BpkPopoverPortal extends Component {
+export type Props = {
+  ...$Exact<PopoverProps>,
+  target: Function | Node,
+  isOpen: boolean,
+  onClose: Function,
+  tetherOptions: Object,
+  portalStyle: ?HTMLElement,
+  portalClassName: ?string,
+  renderTarget: ?Function,
+};
+
+class BpkPopoverPortal extends Component<Props> {
+  tether: ?Tether;
+
+  static propTypes = {
+    ...BpkPopover.propTypes,
+    target: PropTypes.oneOfType([PropTypes.func, PropTypes.node]).isRequired,
+    isOpen: PropTypes.bool.isRequired,
+    onClose: PropTypes.func.isRequired,
+    tetherOptions: PropTypes.shape({
+      attachment: PropTypes.string,
+      targetAttachment: PropTypes.string,
+      offset: PropTypes.string,
+      constraints: PropTypes.array,
+    }),
+    portalStyle: PropTypes.object, // eslint-disable-line react/forbid-prop-types
+    portalClassName: PropTypes.string,
+    renderTarget: PropTypes.func,
+  };
+
+  static defaultProps = {
+    ...BpkPopover.defaultProps,
+    tetherOptions: {
+      attachment: 'top center',
+      constraints: [
+        {
+          to: 'window',
+          attachment: 'together',
+          pin: true,
+        },
+      ],
+    },
+    portalStyle: null,
+    portalClassName: null,
+    renderTarget: null,
+  };
+
   constructor() {
     super();
 
     this.tether = null;
-
-    this.onRender = this.onRender.bind(this);
-    this.beforeClose = this.beforeClose.bind(this);
   }
 
-  onRender(popoverElement, targetElement) {
+  onRender = (popoverElement: HTMLElement, targetElement: ?HTMLElement) => {
     this.position(popoverElement, targetElement);
-  }
+  };
 
-  beforeClose(done) {
+  beforeClose = (done: Function) => {
```

Comment:
We can have a better type here

Reply:
Ended up with `beforeClose = (done: () => void) => {`